inc/contenttemplates/parameters/abstractparameters.class.php

```diff
@@ -53,7 +53,7 @@ abstract class AbstractParameters implements TemplatesParametersInterface
     *
     * @return \Glpi\ContentTemplates\Parameters\ParametersTypes\AbstractParameterType[]
     */
-   abstract protected function defineParameters(): array;
+   abstract public function defineParameters(): array;
```

Comment:
Changing it to public will result in exposing two methods that are almost doint the same thing, which may lead to bad usage of both of them.

I think that `ObjectParameter::compute()` should take care of parameters computation, with something like this:
```php
   public function compute(): array {
      return [
         'type'       => "ObjectParameter",
         'key'        => $this->key,
         'label'      => $this->template_parameters->getObjectLabel(),
         'properties' => TemplateManager::computeParameters($this->template_parameters->getAvailableParameters()),
      ];
   }
```

By doing this, we would be able to replace all `public function defineParameters(): array` to `public function getAvailableParameters(): array`, and use this method in `TemplateDocumentation::addSection()`.

Reply:
Done c9a14bd914.